src/circuit/gadget/sinsemilla/note_commit.rs

```diff
@@ -1058,353 +1113,276 @@ impl NoteCommitConfig {
+        let d_0 = layouter.assign_region(
+            || "NoteCommit MessagePiece d",
+            |mut region| {
+                self.q_notecommit_d.enable(&mut region, 0)?;
 
-                    // advices[6]
-                    copy(
-                        &mut region,
-                        || "b3_c_prime",
-                        self.advices[6],
-                        offset,
-                        &gate_cells.b3_c_prime,
+                copy(&mut region, || "d", col_l, 0, &gate_cells.d)?;
+                let d_0 = {
+                    let cell = region.assign_advice(
+                        || "d_0",
+                        col_m,
+                        0,
+                        || gate_cells.d_0.ok_or(Error::SynthesisError),
                     )?;
+                    CellValue::new(cell, gate_cells.d_0)
+                };
+                copy(&mut region, || "d_1", col_r, 0, &gate_cells.d_1)?;
 
-                    // advices[7]
-                    copy(&mut region, || "c", self.advices[7], offset, &gate_cells.c)?;
+                copy(&mut region, || "d_2", col_m, 1, &gate_cells.d_2)?;
+                copy(&mut region, || "d_3 = z1_d", col_r, 1, &gate_cells.z1_d)?;
 
-                    // advices[8]
-                    copy(&mut region, || "a", self.advices[8], offset, &gate_cells.a)?;
+                Ok(d_0)
+            },
+        )?;
 
-                    // advices[9]
-                    copy(
-                        &mut region,
-                        || "gd_x",
-                        self.advices[9],
-                        offset,
-                        &gate_cells.gd_x,
-                    )?;
-                }
+        // | A_6 | A_7 | A_8 | q_notecommit_e |
+        // ------------------------------------
+        // |  e  | e_0 | e_1 |       1        |
+        layouter.assign_region(
+            || "NoteCommit MessagePiece e",
+            |mut region| {
+                self.q_notecommit_e.enable(&mut region, 0)?;
 
-                // Offset 2
-                {
-                    let offset = 2;
+                copy(&mut region, || "e", col_l, 0, &gate_cells.e)?;
+                copy(&mut region, || "e_0", col_m, 0, &gate_cells.e_0)?;
+                copy(&mut region, || "e_1", col_r, 0, &gate_cells.e_1)?;
 
-                    // advices[0]
-                    copy(
-                        &mut region,
-                        || "e_1",
-                        self.advices[0],
-                        offset,
-                        &gate_cells.e_1,
-                    )?;
+                Ok(())
+            },
+        )?;
 
-                    // advices[1]
-                    copy(&mut region, || "f", self.advices[1], offset, &gate_cells.f)?;
+        // | A_6 | A_7 | q_notecommit_g |
+        // ------------------------------
+        // |  g  | g_0 |       1        |
+        // | g_1 | g_2 |       0        |
```

Comment:
Since the `NoteCommit` config uses four advice columns, we could probably flatten this check into one row.